src/server/auditserver/ldcs_elf_read.c

```diff
@@ -32,10 +32,9 @@ static int readUpTo(FILE *f, unsigned char *buffer, size_t *cur_pos, size_t new_
       return 0;
 
    do {
+      errno = 0;
       result = fread(buffer + *cur_pos, 1, new_size - *cur_pos, f);
-   } while (result == -1 && errno == EINTR);
-   if (result == -1)
-      return -1;
+   } while (errno == EINTR);
```

Comment:
Needs the (result == -1) part of the check. errno only has meaning if the function returns -1

Reply:
Weirdly enough, `fread()` doesn't return -1 on error, and doesn't set `errno`.  That might not be what you want.